engine/src/main/java/io/zeebe/engine/state/variable/DbVariableState.java

```diff
@@ -370,25 +319,22 @@ public DirectBuffer getVariablesAsDocument(
 
   @Override
   public DirectBuffer getVariablesLocalAsDocument(final long scopeKey) {
-
     writer.wrap(documentResultBuffer, 0);
-
     writer.reserveMapHeader();
 
-    variableCount = 0;
-
+    final MutableInteger variableCount = new MutableInteger();
```

Comment:
The variable needs to be either final, or an instance field. So before we had an instance field and it could be primitive - however now if I use a variable, it cannot be a primitive since a final primitive is immutable.

imo it's silly to have a field just for this, and a `MutableInteger` works just as well.